cmake/modules/Findcrypto.cmake

```diff
@@ -56,7 +56,10 @@ else()
     )
 
     if (NOT crypto_LIBRARY)
-        if (BUILD_SHARED_LIBS)
```

Comment:
Could you expand a little bit on the use case? A customer wants to link against `libcrypto.so`, but they don't want to set `-DBUILD_SHARED_LIBS=ON`. Why don't they want to toggle that value? Does it have some side-effects that they are trying to avoid?

Reply:
At a high level "customer wants to statically link curl but dynamically link openssl".

Further explaining the dependency tree
```
SDK-- Curl - Libcrypto
        |- CRT - Libcrypto
         |- S2N - Libcrypto
```
So a user upstream wishes to statically link their curl library (which is completely unrelated to you) so they set `BUILD_SHARED_LIBS` to OFF. Which is what they should be doing. So that path ends up happy. but when we get to `Findcrypto` in the tree it choose static if present. Which is not what they want, but they cannot swap `BUILD_SHARED_LIBS` because they need it for other static dependencies.

[Gtest does something similar for their CRT](https://github.com/google/googletest/blob/main/googletest/CMakeLists.txt#L11-L16)
```
# When other libraries are using a shared version of runtime libraries,
# Google Test also has to use one.
option(
  gtest_force_shared_crt
  "Use shared (DLL) run-time lib even when Google Test is built as static lib."
  OFF)
```

[Also we actually do this already for our CRT](https://github.com/aws/aws-sdk-cpp/blob/main/CMakeLists.txt#L34)
```
option(FORCE_SHARED_CRT "If enabled, will unconditionally link the standard libraries in dynamically, otherwise the standard library will be linked in based on the BUILD_SHARED_LIBS setting" ON)
```